formats/gltf/structure.go

```diff
@@ -27,6 +27,26 @@ type ChildOfRootProperty struct {
 	Name string `json:"name,omitempty"`
 }
 
+func (s ChildOfRootProperty) equal(other ChildOfRootProperty) bool {
+	if s.Name != other.Name || len(s.Extensions) != len(other.Extensions) || len(s.Extras) != len(other.Extras) {
+		return false
+	}
+
+	for key, val := range s.Extensions {
+		if other.Extensions[key] != val {
+			return false
+		}
+	}
```

Comment:
There is a general bug in this code due to the fact that `map[thing]` will return the default value for the value type of map. In this case, nil. We need to make sure that `other.Extras` contains the key as well. Or we can get some false positive situations

It might be worth considering the use of `reflect.DeepEqual` for comparisons of all map types. If people are truly concerned about performance, they can always take the time to satisfy the pointer shortcut we provide.

https://go.dev/play/p/TnjBO6Ub-yp
```go
package main

import "fmt"

func equal(a, b map[string]any) bool {
	if len(a) != len(b) {
		return false
	}

	for key, val := range a {
		if b[key] != val {
			return false
		}
	}

	return true
}

func main() {
	a := map[string]any{
		"a": 1,
		"b": nil,
	}
	b := map[string]any{
		"a": 1,
		"c": 2,
	}

	fmt.Println(equal(a, b)) // Prints true
}

```

Reply:
Ah, that one is easy to fix, I've pushed an update. 

I'm somewhat surprised that `any == any` comparison works at all to be honest, and I don't really like this `map[string]any` type shape, it's very amorphous. I've pushed #52 as an example of how extensions could be made typed and explicitly comparable. This would avoid having `any` there at all, and still leave it flexible enough to permit custom extensions. They would just need to implement the interface.